controller-client/src/lib.rs

```diff
@@ -310,10 +313,33 @@ async fn get_channel(config: &ClientConfig) -> Channel {
             "getting channel for uri {:?} with controller TLS enabled",
             uri_result
         );
-        let pem = std::fs::read(&config.trustcert).expect("read truststore");
-        let ca = Certificate::from_pem(pem);
+        let mut rustls_client_config = RustlsClientConfig::new();
+        rustls_client_config.alpn_protocols.push(Vec::from("h2"));
+        let cert_paths =
+            std::fs::read_dir(&config.trustcert).expect("cannot read from the provided cert directory");
+        for entry in cert_paths {
+            let path = entry.expect("get the cert file").path();
+            debug!("reading cert file {}", path.display());
+            let mut pem = BufReader::new(File::open(path.clone()).expect("read cert file"));
```

Comment:
Two things:
1. This is doing blocking IO in an async method.
2. This is done more than once: `reset` and `refresh_token_if_needed` both call this. - It would be much better if this were done once in the lifetime of the controller-client.